libpod/runtime_img.go

```diff
@@ -1262,3 +1266,219 @@ func getPolicyContext(ctx *types.SystemContext) (*signature.PolicyContext, error
 	}
 	return policyContext, nil
 }
+
+// sizer knows its size.
+type sizer interface {
+	Size() (int64, error)
+}
+
+func imageSize(img types.ImageSource) *uint64 {
+	if s, ok := img.(sizer); ok {
+		if sum, err := s.Size(); err == nil {
+			usum := uint64(sum)
+			return &usum
+		}
+	}
+	return nil
+}
+
+func reposToMap(repotags []string) map[string]string {
+	// map format is repo -> tag
+	repos := make(map[string]string)
+	for _, repo := range repotags {
+		var repository, tag string
+		if len(repo) > 0 {
+			li := strings.LastIndex(repo, ":")
+			repository = repo[0:li]
+			tag = repo[li+1:]
+		}
+		repos[repository] = tag
+	}
+	if len(repos) == 0 {
+		repos["<none>"] = "<none"
+	}
+	return repos
+}
+
+// GetImagesForImages gets the images for podman images and returns them as
+// an array of ImageResults
+func (r *Runtime) GetImagesForImages() ([]inspect.ImageResult, error) {
+	var results []inspect.ImageResult
+
+	foo, err := r.store.Images()
+	if err != nil {
+		return nil, err
+	}
+	for _, image := range foo {
+		bar, err := is.Transport.ParseStoreReference(r.store, image.ID)
+		if err != nil {
+			return nil, err
+		}
+		systemContext := &types.SystemContext{}
+		img, err := bar.NewImageSource(systemContext)
+		if err != nil {
+			return nil, err
+		}
+		ic, err := bar.NewImage(&types.SystemContext{})
+		if err != nil {
+			return nil, err
+		}
+		imgInspect, err := ic.Inspect()
+		if err != nil {
+			return nil, err
+		}
+		dangling := false
+		if len(image.Names) == 0 {
+			dangling = true
+		}
+
+		for repo, tag := range reposToMap(image.Names) {
```

Comment:
Would it make sense to have a map of repo to tag in the struct instead of here? Repeatedly adding the same image seems kind of gross. Would make our output logic that much uglier, though... Ew.

Reply:
lets hold off on this for now. the list is getting long